airflow/providers/jenkins/operators/jenkins_job_trigger.py

```diff
@@ -19,19 +19,22 @@
 import json
 import socket
 import time
+from typing import Any, Mapping, Optional, Type, Union
 from urllib.error import HTTPError, URLError
 
 import jenkins
-from jenkins import JenkinsException
+from jenkins import Jenkins, JenkinsException
 from requests import Request
 
 from airflow.exceptions import AirflowException
 from airflow.models import BaseOperator
 from airflow.providers.jenkins.hooks.jenkins import JenkinsHook
 from airflow.utils.decorators import apply_defaults
 
+JReqType = Mapping[str, Union[str, Mapping[str, str]]]
 
-def jenkins_request_with_headers(jenkins_server, req):
+
+def jenkins_request_with_headers(jenkins_server: Type[Jenkins], req: Request) -> JReqType:
```

Comment:
If the result is always in type of `{'body': ..., 'headers': ...}` I would suggest introducing named tuple. This should solve a lot of problems and will add some useful stuff like accessing using attributes.
```python
class JenkingRequest(NamedTuple):
    body: str
    headers: Dict[str, str]
```

Reply:
The headers have been a little difficult to pin down, so I went with a simpler approach.